### PR TITLE
fetch Catch2 and update include statements

### PR DESCRIPTION
**Pull request by LoreMoretti**
This PR aims to address https://github.com/robotology/walking-controllers/issues/183

### CODE REVIEW

**Review by S-Dafarra — APPROVED** [review]
Thanks a lot for taking care of this @LoreMoretti!

File: cmake/FetchCatch2.cmake [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+if(NOT USE_SYSTEM_Catch2)
+  include(FetchContent)
+  FetchContent_Declare(Catch2
+    GIT_REPOSITORY https://github.com/catchorg/Catch2.git
+    GIT_TAG        v3.0.1)
+  FetchContent_GetProperties(Catch2)
+  if(NOT Catch2_POPULATED)
+    message(STATUS "Fetching Catch2...")
+    FetchContent_MakeAvailable(Catch2)
+  endif()
```
Comment (traversaro) [inline cmake/FetchCatch2.cmake:23]: I think you can directly call `FetchContent_MakeAvailable(Catch2)`, see https://cmake.org/cmake/help/latest/module/FetchContent.html .

Comment (S-Dafarra) [reply cmake/FetchCatch2.cmake:23]: Actually, I think that wit the additional ``if`` we can also have the message when ``Catch2`` is downloaded, right?

Comment (traversaro) [reply cmake/FetchCatch2.cmake:23]: > Actually, I think that wit the additional ``if`` we can also have the message when ``Catch2`` is downloaded, right?

Good point, I did not thought of that!

**Pull request merged**